### PR TITLE
Reproduce data from data-raw and test on CI (#1)

### LINKED ISSUE #61: Test data-raw/* produces the expected output
The contents of data/ come from computations we do in data-raw/. We have regression tests for the contents of data/ but we do not have tests to ensure all scripts in data-raw/* can be sourced and result in the expected output in data/. Such tests should be part of our CI workflow. I see to options:
* We hack a GitHub action to source all files in data-raw/ and compare the output against data/.
* Or we extract all we now do in data-raw/ into its own package and we use standard tests and CI. This package would not be on CRAN. (The name r2dii.dataraw is taken. We may use instead the name r2dii.raw).

### PR DESCRIPTION
**Pull request by maurolepore**
Closes #61 
https://github.com/2DegreesInvesting/r2dii.data/issues/76

* New internal `source_data_raw()` helps source all .R files 
  under data-raw/.
* New `expect_no_differences()` fails with informative output
  of `waldo::compare()`.
* CI workflow reproduces all data/ from data-raw/ and checks
  for differences agains regression references.

For example: 

Say I change something in data-raw/. I don't source any file nor test or check locally. I just commit and push. Now our GitHub Actions workflow sources all files in data-raw/, which updates all files in data/. As usual, R CMD check runs all tests and one of them fails because the updated dataset is now different that the regression reference scored in tests/testthat/ref-*.

A failing test on GitHub Actions now shows not only this usual uninformative result:

image

but also this more informative result:

image

This should allow us to fix the problem more quickly.

### CODE REVIEW

**Comment by jdhoffa** [comment]
A couple of points: 
- I see failing checks in CI, which seem to occur when sourcing data-raw. In particular, we do not list the package `here` under Imports, but it is used a lot in "data-raw/*.R" files. 
- Running `source_data_raw()` locally on my machine errors, stating that it cannot find `enlist_datasets()`

**Comment by maurolepore** [comment]
> Running source_data_raw() locally on my machine errors, stating that it cannot find enlist_datasets()

Those are all internal functions. Did you `devtools::load_all()`?

**Comment by maurolepore** [comment]
> I see failing checks in CI, which seem to occur when sourcing data-raw. In particular, we do not list the package here under Imports, but it is used a lot in "data-raw/*.R" files.

We don't need to Import packages we use in data-raw/. To avoid the error on CI you can install those packages on the GitHub Actions workflow file. I did just that for dplyr and readr, but removed here because it's needless.

File: data-raw/sector_classifications.R [hunk 1]
```diff
@@ -13,7 +13,7 @@ source(file.path("data-raw", "utils.R"))
 #'   at [data_dictionary].
 #' @noRd
 sector_classification_df <- function() {
-  out <- enlist_datasets("r2dii.data", pattern = "_classification$")
+  out <- r2dii.data:::enlist_datasets("r2dii.data", pattern = "_classification$")
```
[the remark this answers is not in the document]
Comment (maurolepore) [reply data-raw/sector_classifications.R]: I placed in in R/ so I can use it in tests/thestthat/test-all.R:

```r
test_that("no data inherits spec_tbl_df", {
  library(r2dii.data)
  has_spec <- vapply(
    enlist_datasets("r2dii.data"),
    function(.x) inherits(.x, "spec_tbl_df"),
    logical(1)
  )

  expect_false(any(has_spec))
})
```

**Pull request merged**